specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/preview/2023-05-01-preview/migrate.json

```diff
@@ -17936,6 +18241,27 @@
         }
       }
     },
+    "ErrorDetail": {
+      "type": "object",
+      "description": "Error Details",
+      "properties": {
+        "errorCode": {
+          "type": "integer",
+          "format": "int32",
```

Comment:
@pratikjoshi14 The main purpose of introducing error code is to support localization of error messages. This is the requirement from the UI side that the API should send the error code because they maintain some kind of mapping between the error code and the error message. The same approach is being used by other services as well.

By convention, Error Code should always be int. Why do you think it should be string ? Also, since we are also sending the respective error message along with error code in the response, i don't think customers will face any issue in interpreting them. If we would have sent only error code, then it would have made sense to document them.

Reply:
It could have been string because at other places and even in ARM common types, error code is a string:  https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v6/types.json
Hence thought of checking ones.

Also did not understand this part: "The main purpose of introducing error code is to support localization of error messages. This is the requirement from the UI side that the API should send the error code because they maintain some kind of mapping between the error code and the error message. ".. if this is UI side mapping, how can it be handled for other types of clients like PS/CLI/SDK?

Reply:
Ok.